### ISSUE TITLE
NWjs update

### ISSUE THREAD

**Issue by pcenta** [popcorn-official/popcorn-desktop#2062]
**Is your feature request related to a problem? Please describe.**
Not related to a problem but the project is using outdated version.

**Describe the idea you'd like**
App loads NWjs from https://get.popcorntime.app/repo/nw/. First step would be to upload newer version to allow easier testing than manually tricking app on local build.

**Describe alternatives you've considered**
We can stay on existing version?

**Additional context**
NWjs link https://nwjs.io/

**Comment by kiriles90** [comment]
The idea of updating nwjs is great (and will be necessary again at some point) but its not without issues. Mainly that someone has to compile the new version to include the extra codecs then test/fix anything that broke, which will probably happen since we are bumping almost 10 major releases.

Do you want me to tag this as a Feature and we see what happens? Maybe we can make a new branch if someone(s) wants to take this up so as to not break the dev builds until the new nwjs one gets ironed out.

**Comment by pcenta** [comment]
Locally I tried it with v0.52.0 and I've tested some basic functionalities and haven't found anything broken. Since our current version works there's no hurry to do this but I think that there are some benefits (recent memory usage optimizations).
Even though this is almost 10 releases newer I don't think that there will be any serious issues with it. Majority of those releases are only bumped versions of Chromium and Node and I don't think that any of them recently deprecated or removed anything PT uses.

If you could tag this as a Feature would be great. Could you upload new NWjs SDK too? Anyone can do this but it would be easier if anyone deletes their cache folders. From there it's all about changing version defined in one of the source files.

**Comment by kiriles90** [comment]
> Could you upload new NWjs SDK too?  

I haven't tried to compile the extra codecs in and tbh the comments from whoever tried was discouraging enough for me not even bothering attempting it but @team-pct might know what we need to do?

**Comment by kiriles90** [comment]
> Could you upload new NWjs SDK too?  

I've never tried compiling the extra codecs in (and tbh the comments from whoever tried was discouraging enough for me not even bothering finding out more) but @team-pct might know what exactly we need to do?

**Comment by Ouack23** [comment]
Hello,
I am curious about how you made https://nwjs.io/ work in PT.
I tried to use the following patch for gulpfile.js :
```
--- src/popcorn-desktop/gulpfile.js     2021-04-13 12:28:56.098245263 +0200
+++ gulpfile.js 2021-04-13 12:27:57.132266951 +0200
@@ -142,8 +142,7 @@
   macIcns: './src/app/images/butter.icns',
   version: nwVersion,
   flavor: nwFlavor,
-  manifestUrl: 'https://popcorntime.app/version.json',
-  downloadUrl: 'https://get.popcorntime.app/repo/nw/',
+  downloadUrl: 'https://dl.nwjs.io/',
   platforms: parsePlatforms()
 }).on('log', console.log);
```

PT successfully builds but do not launch (no clear log is output).
How did you do that ?

Thanks for reading

**Comment by pcenta** [comment]
> Hello,
> I am curious about how you made https://nwjs.io/ work in PT.
> I tried to use the following patch for gulpfile.js :
> 
> ```
> --- src/popcorn-desktop/gulpfile.js     2021-04-13 12:28:56.098245263 +0200
> +++ gulpfile.js 2021-04-13 12:27:57.132266951 +0200
> @@ -142,8 +142,7 @@
>    macIcns: './src/app/images/butter.icns',
>    version: nwVersion,
>    flavor: nwFlavor,
> -  manifestUrl: 'https://popcorntime.app/version.json',
> -  downloadUrl: 'https://get.popcorntime.app/repo/nw/',
> +  downloadUrl: 'https://dl.nwjs.io/',
>    platforms: parsePlatforms()
>  }).on('log', console.log);
> ```
> 
> PT successfully builds but do not launch (no clear log is output). I tested 0.44.5, 0.52.0 and 0.52.2 and none of them worked.
> How did you do that ?
> 
> Thanks for reading
> 
> Edit : from what I understand, I have to include support at compile time for various proprietary codecs in NWJS. Still, it is unclear to me what steps I have to setup for a complete compile script.

What've done is a bit different. I manually downloaded NWjs SDK and put it into my cache folder since I wanted minimal change in actual code. Also version that worked for me was `0.52.0`, it might be that there were some additional changes in other versions that broke something?

**Comment by chengsokdara** [comment]
I removed those two lines without providing new downloadUrl.
```
--- src/popcorn-desktop/gulpfile.js     2021-04-13 12:28:56.098245263 +0200
+++ gulpfile.js 2021-04-13 12:27:57.132266951 +0200
@@ -142,8 +142,7 @@
   macIcns: './src/app/images/butter.icns',
   version: nwVersion,
   flavor: nwFlavor,
-  manifestUrl: 'https://popcorntime.app/version.json',
-  downloadUrl: 'https://get.popcorntime.app/repo/nw/',
   platforms: parsePlatforms()
 }).on('log', console.log);
```

The build fallback to download by itself, seem nwjs-sdk-v0.44.5 works perfectly.
```
Create cache folder in /Users/ra/Projects/popcorn-official/popcorn-desktop/cache/0.44.5-sdk
Downloading: https://dl.nwjs.io/v0.44.5/nwjs-sdk-v0.44.5-osx-x64.zip
```

**Comment by memarko** [comment]
Thanks it works.

I have problems accessing https://get.popcorntime.app/repo/nw/ is this domain down or is it just blocked by my ISP?

> I removed those two lines without providing new downloadUrl.
> 
> ```
> --- src/popcorn-desktop/gulpfile.js     2021-04-13 12:28:56.098245263 +0200
> +++ gulpfile.js 2021-04-13 12:27:57.132266951 +0200
> @@ -142,8 +142,7 @@
>    macIcns: './src/app/images/butter.icns',
>    version: nwVersion,
>    flavor: nwFlavor,
> -  manifestUrl: 'https://popcorntime.app/version.json',
> -  downloadUrl: 'https://get.popcorntime.app/repo/nw/',
>    platforms: parsePlatforms()
>  }).on('log', console.log);
> ```
> 
> The build fallback to download by itself, seem nwjs-sdk-v0.44.5 works perfectly.
> 
> ```
> Create cache folder in ~/Projects/popcorn-official/popcorn-desktop/cache/0.44.5-sdk
> Downloading: https://dl.nwjs.io/v0.44.5/nwjs-sdk-v0.44.5-osx-x64.zip
> ```

**Comment by ivan1986** [comment]
@memarko domain is down
if you use https://dl.nwjs.io you get version without proprietary codecs
i also save copy of 0.44.5

manifestUrl: 'http://popcorn-ru.tk/version.json',
downloadUrl: 'http://popcorn-ru.tk/nw/',

**Comment by isaamos** [comment]
> 
> 
> @memarko domain is down
> if you use https://dl.nwjs.io you get version without proprietary codecs
> i also save copy of 0.44.5
> 
> manifestUrl: 'http://popcorn-ru.tk/version.json',
> downloadUrl: 'http://popcorn-ru.tk/nw/',

Is this reliable or secure?

**Comment by ivan1986** [comment]
@isaamos it's safe

**Comment by pcenta** [comment]
> @memarko domain is down
> if you use https://dl.nwjs.io you get version without proprietary codecs
> i also save copy of 0.44.5
> 
> manifestUrl: 'http://popcorn-ru.tk/version.json',
> downloadUrl: 'http://popcorn-ru.tk/nw/',

FFmpeg can be added manually to SDK or SDK itself can be built with included codec (it's an option if you build NWjs manually). So this means we can download NWjs from https://nwjs.io/downloads/ and add prebuilt FFmpeg codec from https://github.com/iteufel/nwjs-ffmpeg-prebuilt. I've tested it locally with NWjs 54.1 and prebuilt FFmpeg for 54. 

Can someone else test this and report if there are any issues/benefits with newer version of Chromium?

**Comment by ivan1986** [comment]
For me build with external ffmpeg not work - many films without sound
@team-pct said that he has a build script

**Comment by ivan1986** [comment]
i commit current build script to
https://github.com/popcorn-time-ru/nwjs-build
it build nwjs with codecs, but some incorrect flags
eac3 (a52) audio not working and avi files not play

**Comment by Kief5555** [comment]
> I removed those two lines without providing new downloadUrl.
> 
> ```
> --- src/popcorn-desktop/gulpfile.js     2021-04-13 12:28:56.098245263 +0200
> +++ gulpfile.js 2021-04-13 12:27:57.132266951 +0200
> @@ -142,8 +142,7 @@
>    macIcns: './src/app/images/butter.icns',
>    version: nwVersion,
>    flavor: nwFlavor,
> -  manifestUrl: 'https://popcorntime.app/version.json',
> -  downloadUrl: 'https://get.popcorntime.app/repo/nw/',
>    platforms: parsePlatforms()
>  }).on('log', console.log);
> ```
> 
> The build fallback to download by itself, seem nwjs-sdk-v0.44.5 works perfectly.
> 
> ```
> Create cache folder in ~/Projects/popcorn-official/popcorn-desktop/cache/0.44.5-sdk
> Downloading: https://dl.nwjs.io/v0.44.5/nwjs-sdk-v0.44.5-osx-x64.zip
> ```

How did you download it on mac? Its not opening for me

**Comment by Kief5555** [comment]
@kiriles90, you should use electron to build this app. https://www.electronjs.org/

**Comment by Kief5555** [comment]
> Locally I tried it with v0.52.0 and I've tested some basic functionalities and haven't found anything broken. Since our current version works there's no hurry to do this but I think that there are some benefits (recent memory usage optimizations). Even though this is almost 10 releases newer I don't think that there will be any serious issues with it. Majority of those releases are only bumped versions of Chromium and Node and I don't think that any of them recently deprecated or removed anything PT uses.
> 
> If you could tag this as a Feature would be great. Could you upload new NWjs SDK too? Anyone can do this but it would be easier if anyone deletes their cache folders. From there it's all about changing version defined in one of the source files.

How did you test?

**Comment by Kief5555** [comment]
> I removed those two lines without providing new downloadUrl.
> 
> ```
> --- src/popcorn-desktop/gulpfile.js     2021-04-13 12:28:56.098245263 +0200
> +++ gulpfile.js 2021-04-13 12:27:57.132266951 +0200
> @@ -142,8 +142,7 @@
>    macIcns: './src/app/images/butter.icns',
>    version: nwVersion,
>    flavor: nwFlavor,
> -  manifestUrl: 'https://popcorntime.app/version.json',
> -  downloadUrl: 'https://get.popcorntime.app/repo/nw/',
>    platforms: parsePlatforms()
>  }).on('log', console.log);
> ```
> 
> The build fallback to download by itself, seem nwjs-sdk-v0.44.5 works perfectly.
> 
> ```
> Create cache folder in ~/Projects/popcorn-official/popcorn-desktop/cache/0.44.5-sdk
> Downloading: https://dl.nwjs.io/v0.44.5/nwjs-sdk-v0.44.5-osx-x64.zip
> ```

Can you provide me with a download link to the working version? (I am on mac)

**Comment by kiriles90** [comment]
@Kief5555 An easy way to build without installing any build tools or libraries or having to learn anything is to just fork the dev branch, enable 'Actions' in your branch, open `gulpfile.js` for edit, remove those two lines about the custom nwjs above and save the file. A new commit should have been merged and new builds for all OSes should start building in the Actions page, wait about 15 minutes and download the macOS one once its done.  

(mentioning this so you dont have to rely on someone else for builds every time there is an update if you are going to try using the latest nwjs without codecs)

**Comment by Kief5555** [comment]
> @Kief5555 An easy way to build without installing any build tools or libraries or having to learn anything is to just fork the dev branch, enable 'Actions' in your branch, open `gulpfile.js` for edit, remove those two lines about the custom nwjs above and save the file. A new commit should have been merged and new builds for all OSes should start building in the Actions page, wait about 15 minutes and download the macOS one once its done.
> 
> (mentioning this so you dont have to rely on someone else for builds every time there is an update if you are going to try using the latest nwjs version without codecs)

Hey, I did it a different way. I bumped the NWJS version to 0.55 in gulpfile.js then removed the 2 lines where it takes downloads from. I built it using instructions in readme, but did not start. Then I downloaded Ivan's audio fixed and put it in builds then ran it. Ran just fine on Monterey.

**Comment by Kief5555** [comment]
https://youtu.be/3IZid6lBMJM I have a fix, you can watch from the link.

**Comment by ivan1986** [comment]
@Kief5555 you write this 5 times
try to read thread
it's not work - many videos have not sound

**Comment by Kief5555** [comment]
> @Kief5555 you write this 5 times try to read thread it's not work - many videos have not sound

some videos do, wont they have sound if they have the codecs?

**Comment by ivan1986** [comment]
@Kief5555 need rebuild full nwjs with enable codecs patch
i try to do this - https://github.com/popcorn-official/popcorn-desktop/issues/2062#issuecomment-932817059
it works better, and play many films with sound, but ac3 audio not work, some icorrct flags
if you want - try to build and test this on your mac - it will be more useful than a video that can be replaced with a couple of paragraphs of text

**Comment by Kief5555** [comment]
@ivan1986 Do you have the link to downloaded these codecs?

**Comment by ivan1986** [comment]
what link?
this codecs is part of ffmpeg, need compile this with enable codecs
https://github.com/popcorn-time-ru/nwjs-build/blob/master/build_linux64.sh#L73-L108
i apply patch after download sources

**Comment by danielbayley** [comment]
@ivan1986 Can we work these steps into the Homebrew [Cask](https://github.com/popcorn-official/popcorn-desktop/blob/development/casks/popcorn-time.rb) if they produce a working build on the latest macOS…?

**Comment by ivan1986** [comment]
@danielbayley Need build last nwjs with patches for all systems, for latest mac, win and linux - not need use homebrew for it, it's installer.
I try build by this manual
https://docs.nwjs.io/en/latest/For%20Developers/Building%20NW.js/
https://docs.nwjs.io/en/latest/For%20Developers/Enable%20Proprietary%20Codecs/

**Comment by danielbayley** [comment]
@pcenta @ivan1986 @Kief5555 See #2439 where I have tried to address this… It produces a working build for me on macOS 12.2.1, and is based on these [steps](https://github.com/popcorn-official/popcorn-desktop/wiki/Fix-Crashing-On-MacOS-12#steps) in the wiki.

**Comment by Kief5555** [comment]
> @pcenta @ivan1986 @Kief5555 See #2439 where I have tried to address this… It produces a working build for me on macOS 12.2.1, and is based on these [steps](https://github.com/popcorn-official/popcorn-desktop/wiki/Fix-Crashing-On-MacOS-12#steps) in the wiki.

I will update the wiki when audio fix is ready. For now, only some (very little) movies/shows have working audio

**Comment by danielbayley** [comment]
> > @pcenta @ivan1986 @Kief5555 See #2439 where I have tried to address this… It produces a working build for me on macOS 12.2.1, and is based on these [steps](https://github.com/popcorn-official/popcorn-desktop/wiki/Fix-Crashing-On-MacOS-12#steps) in the wiki.
> 
> I will update the wiki when audio fix is ready. For now, only some (very little) movies/shows have working audio

@Kief5555 Ok cool, ping me when you have it, and I can finish #2439…

**Comment by berard** [comment]
👋 it was suggested that I discuss in this thread a (now closed) PR I made for popcorn-desktop, which combined with one [for nw-builder](https://github.com/nwjs-community/nw-builder/pull/547) popcorn-desktop can compile and run on M1. maybe something to include in this work around nwjs?

**Comment by Ge0rges** [comment]
@berard you should contribute so that it will run on M1 Mac.

**Comment by Kief5555** [comment]
Hmm... nwjs `0.55.0` seems to lag my mac quite a bit, just realized. Espcially when downloading. Having high cpu jumps with only chromium and discord running in the backround, around 40% cpu. 

Device: 2012 MBP Retina 15". Intel i7 2.3 ghz 8GB ram. One of my gpus has a mosfet failure but I am running intel gpu.

**Comment by berard** [comment]
@Ge0rges it is working on M1 here: https://github.com/berard/popcorn-desktop

**Comment by jlokos** [comment]
@berard are you able to release the dist version? Everytime I try to compile the src and use it the package seems to be broken? Only works on device it is compiled on.

**Comment by Ge0rges** [comment]
I second what @jlokos said. I also suggest a pull request if you have a working fork @berard.

**Comment by Kief5555** [comment]
How come I cannot comment?

**Comment by Kief5555** [comment]
_Just use windows:tm:_

**Comment by kiriles90** [comment]
@tio-trom I said before, set the frame flag in package.json manually or make sure it changes when you do it via the app, then you'll 100% know if its nwjs (@Kief5555 and yes using Windows wouldnt be a bad idea, then at least I can debug things for you since no macOS dev/user ever does the same :)

**Comment by tio-trom** [comment]
If you can please tell me where I can find that file. All I found is in the home directory under the settings that are stored for PopcornTime `/home/tio/.config/Popcorn-Time/Default/data/` this file `settings.db` where I see `{"key":"nativeWindowFrame","value":true,"_id":"rt7JMUBaQ3AHuzHq"}`

Is the package.json part of the PopcornTime install folder or is it located under some NW.js specific folder?

**Comment by kiriles90** [comment]
Base install dir, this one [/package.json#L32](https://github.com/popcorn-official/popcorn-desktop/blob/development/package.json#L32)

**Comment by tio-trom** [comment]
Yes I found that file on your repo but I cannot find any such file on my system. I think it should be located in the home folder else the app would not have access to the filesystem without the admin password. Perhaps on Linux the above file I mentioned should be the one forcing any settings for the app. Actually if I remove that file or edit it, it changes the settings for the Popcorn app. So they are stored there....

Maybe I do not understand something and maybe it is different on Linux...

**Comment by kiriles90** [comment]
It doesnt work differently in linux, you are removing your user data by removing those .db, (also dont manually edit them you'll end up bugging them and need to clear your data, those kind of settings you change via /src/app/settings.js etc). But.. I'm not talking about internal app settings, I'm talking about bypassing the app to see if the issue is actually NW.js. So.. find your package.json and leave those alone :)

**Comment by tio-trom** [comment]
Ok found it in `/usr/share/popcorntime/` - the frame was set to `false`. Manually editing it to set as `true` does the job :) . So now, who/what is at fault here?

**Comment by kiriles90** [comment]
> It seems setting up this in the app itself won't change it in the package.json.

See how we could have avoided the last 20 comments in the other issue if you've just asked me after any of them "what the hell do you mean with all that mumbo-jumbo you keep talking about nwjs and package.json's.. " :)  

> So now, who/what is at fault here?

Ehm.. @pcenta (since he is here.. :smile:)

@ivan1986 and chance you can take a look at this?

**Comment by tio-trom** [comment]
> See how we could have avoided the last 20 comments in the other issue if you've just asked me after any of them "what the hell do you mean with all that mumbo-jumbo you keep talking about nwjs and package.json's.. " :)

I do not think it was entirely my fault. Maybe you should have told me to check the package.json file that should be inside the install folder for the app, and check a particular line in that file. In any case, at least we have a smoking gun of some sorts now. Thank you for the help.

**Comment by kiriles90** [comment]
> This is all handled by NW.js. The option just edits the window.frame option in package.json from false to true and requires restarting the app to apply.

From my second comment. Then told you a couple of times more until you got me to the point where I even (rudely) said when you kept not getting it..

> Man.. do you understand what a package.json flag is? You can bring 20 people to test it, that doesnt change who handles which code. Its NOT Popcorn Time code in there, THE END. Set the damn flag manually, bypass Popcorn Time's settings, and see if it works, nothing else to be done about it.

I'm sorry but you keep deflecting the blame where to be perfectly honest because I answer support questions all the time the issue here was that you just didnt read carefully and tried to understand what I was telling you, instead you were bypassing whatever you didnt know what it was and kept pushing just to get your goal which was an 'ok I'll fix it' from me. We can keep this up as long as you want but I'm dont think we'll get anywhere so now that the issue is identified let it be fixed, then download the next build after that. Until then, set the flag manually, you know where it is now.

**Comment by tio-trom** [comment]
I said I am using Linux. If I were you I would have said: Go to` /usr/share/popcorntime/` and check the file `package.json` -  see if the frame is set to true or false. Done. I would have done that in a minute :). 

I too provide a lot of technical help for our Operating System and a lot more for our over 30 online services, but you assumed that I know where popcorntime stores its settings files or what NW.js is. Honestly I understood that NW.js is a different package and I have to check my "system's package.json" file or something like that. I understood that this is not a Popcorn Time issue but a Linux/package issue. This:

> Its NOT Popcorn Time code in there, THE END. Set the damn flag manually, bypass Popcorn Time's settings, and see if it works, nothing else to be done about it.

But anyway. It was a communication failure and now you can blame me, I can blame you, but that changes nothing to solve this bug. I will keep an eye on this issue if I need to debug further.

**Comment by kiriles90** [comment]
>  if you've just asked me after any of them "what the hell do you mean with all that mumbo-jumbo you keep talking about nwjs and package.json's.. 

.. instead of assuming random things (you almost did it afterwards too with the db's :) Anyway, enough for me, goodnight man.

**Comment by tio-trom** [comment]
> instead of assuming random things (you almost did it afterwards too with the db's

Because you are telling me to check my "package.json" file as if I were to know this is the popcorn's time package.json file hehe :). So I tried to debug and see where Popcorn Time stores its settings since I could not find it in my system's files because there are so so many package.json files in the system. That's why I started my comment with "If you can please tell me where I can find that file. " 

Damn anyway, how long are we going to try and find the errors in communication!? I hope you go to sleep and I will too, so we leave this as it is. I had no intention to bring the previous convo into this thread but you started it.

**Comment by kiriles90** [comment]
@tio-trom Just ask which (or what is) package.json then? Or you just open a random one from another app? :) Anyway.. so I've just been informed that, this is supposed to be how things work in linux. You cant edit package.json via the app there since it supposedly doesn't have rights to do so. (also sorry @pcenta for assuming there was an issue with your PR). So, as far as I can see this stay's with the manual flag for Linux for now until someone devices a solution if one is even possible. (no, not setting the flag as default :)

**Comment by kiriles90** [comment]
It a file used by Node (NW.js that Popcorn Time runs on is Node + Chromium basically) to store package information including what modules are needed by the app. More info at https://nodejs.dev/learn/the-package-json-guide

**Comment by tio-trom** [comment]
I have a question that is Linux related: since Linux apps do not have access to the system's files without an admin password, and the package.json file is stored in the system's files (usr/share/popcorntime), then it means the Popcorn's Settings cannot be changed in Linux, in theory, if you are using that file for the job. However I can confirm that a lot of settings are indeed changed for Popcorn Time.

My question is: why can some settings be changed and stored for Popcorn Time, but not other like the "native window decoration"?

**Comment by ivan1986** [comment]
https://github.com/popcorn-official/popcorn-desktop/blob/development/src/app/lib/views/settings_container.js#L620=
it was write without support this case

**Comment by kiriles90** [comment]
@tio-trom I edit above with further explanation. Let me know if something still isnt clear so we avoid a repeat of yesterday :)

**Comment by kiriles90** [comment]
@tio-trom The app has its default options stored in /src/app/settings.js. The one's in package.json are Node and Chromium flags/settings. As for the ones in User Data that you were deleting yesterday, those are your profile settings, anything you change from default value, its stored in there then the app uses that instead of the default one in /src/app/settings.js

**Comment by kiriles90** [comment]
I'm the one who marked it (including yours :) We spammed the NW.js update issue a bit..

**Comment by pcenta** [comment]
> I have a question that is Linux related: since Linux apps do not have access to the system's files without an admin password, and the package.json file is stored in the system's files (usr/share/popcorntime), then it means the Popcorn's Settings cannot be changed in Linux, in theory, if you are using that file for the job. However I can confirm that a lot of settings are indeed changed for Popcorn Time (so it must use the files in the .config - the home directory).
> 
> My question is: why can some settings be changed and stored for Popcorn Time, but not other like the "native window decoration"?

Just to add one more spam message :)
With my limited Linux knowledge I think that access to filesystem depends on multiple factors (if you need to enter password regardless then following stuff is irrelavant). One would be that if you install application with root user - owner of files will be root due to being created by root process. That would cause permission issues if app is being run as a non root process.
One workaround that someone can try on Linux system is to see what happens if you manually change owner/permissions of package.json.

We can continue discussion in a new issue with TLDR summary of comments here if there's any debate left.

@tio-trom @kiriles90

**Comment by kiriles90** [comment]
@pcenta @ivan1986 should know more if this is possible or it would bug something?

And yes permissions basically is the only issue with this. The same problem would be there if you purposely did the same on Windows or any other OS.

**Comment by tio-trom** [comment]
@pcenta The install files are under root...I would not want to "hack" that for this reason. The sane way in Linux, is to store all settings under the `.config` folder in `Home`, where the user has access.

**Comment by kiriles90** [comment]
@tio-trom Again, not "our" settings. We dont get to choose were NW.js and Linux store their stuff, but I'm pretty sure there is a good reason for it :)

**Comment by pcenta** [comment]
> @pcenta The install files are under root...I would not want to "hack" that for this reason. The sane way in Linux, is to store all settings under the `.config` folder in `Home`, where the user has access.

If you try to run PT as a root user then it works? Not that this solves anything but just to confirm that this is the cause.

**Comment by tio-trom** [comment]
> @tio-trom Again, not "our" settings. We dont get to choose where NW.js and Linux store their stuff, but I'm pretty sure there is a good reason for it :)

You can mark this comment as off-topic but just so you know, sounds confusing to me when you are saying "this is not out file" while it is inside the popcorntime install folder :P. That confused me all through our conversation, but is ok I accept that you are saying this is beyond your control.

**Comment by pcenta** [comment]
> @tio-trom Again, not "our" settings. We dont get to choose where NW.js and Linux store their stuff, but I'm pretty sure there is a good reason for it :)

NWjs has setting that configures data dir it uses `--user-data-dir` [NWjs docs](https://docs.nwjs.io/en/latest/References/Command%20Line%20Options/). Linux default is `~/.config/<name-in-manifest>`.

**Comment by kiriles90** [comment]
@tio-trom @pcenta afaik as I know there are ways to install the app without using sudo = no admin privileges, but all involve a process. Things like https://github.com/nvm-sh/nvm might help and if I remember there was a guide somewhere in the old subreddit but goodluck finding that archived somewhere. Then.. you just run the app like on Windows, no admin privileges for it or package.json required.

**Comment by kiriles90** [comment]
@tio-trom Sorry I just saw about the being confused about the 'not our setting' thing. NW.js is the framework that Popcorn Time runs on, which in its turn runs on Linux which is the OS. Very simplistically imagine Popcorn Time as a very big website in mostly javascript, css and html that comes prepackaged with Chrome (Chromium) and you install it. The same can be basically said for all apps that use NW.js and all apps that use a framework in general, its the something that runs on something. Even ones that don't and compile/write their own executables, they 99% of the time write it in some high level language, with some libraries, which you cannot escape the constraints and capabilities of. Or.. just write in assembly, and take 3 months to write a calculator app (for which you'll get payed in gold for though, good/fast assembly developers are highly prized :)

**Comment by kiriles90** [comment]
Btw, if anyone wants to continue this or another conversation, now or in the future, just open a new issue, @ whoever you want to see it, close it, then do whatever you want with it. In this one every time we comment everyone that is subbed gets a notification an they might not want to, including the off-topic conversation part..

**Comment by Kief5555** [comment]
> Btw, if anyone wants to continue this or another conversation, now or in the future, just open a new issue, @ whoever you want to see it, close it, then do whatever you want with it. In this one every time we comment everyone that is subbed gets a notification an they might not want to, including the taking the issue off-topic part..

I agree lol

**Comment by Kief5555** [comment]
@kiriles90 Sorry for late response. I will edit docs tomorrow. Thanks for opening it.


Sent from my iPhone

**Comment by kiriles90** [comment]
@Kief5555 no worries, just let me know when you are done so I can re-lock them. (a bit too lazy to check everyday in case someone defaced them or is trying to redirect users to a virus or something, both have already been attempted so better be locked and just unlock for whoever wants to edit them)

**Comment by Kief5555** [comment]
> 

Done @kiriles90

**Comment by kiriles90** [comment]
So, since the development branch has been switched to the latest NW.js build (https://github.com/popcorn-official/popcorn-desktop/commit/30ab2c5dd4177bd804a0a0ecab13e8bb77835763) but still lacks full proprietary audio codec support (the ffmpeg prebuild is better than nothing but is still lacking) I'm changing the title of this issue to reflect the new issue at hand and tagging this as a bug from feature that it was before the update.  

Also.. the old custom 0.44.5 NW.js version is now at [popcorn-official/popcorn-desktop/tree/development_old_nwjs](https://github.com/popcorn-official/popcorn-desktop/tree/development_old_nwjs) though this should be eventually removed once this issue here is resolved/closed. Last, Actions have been updated to reflect the change with now two workflows, one for new and one for old NW.js branches.

**Comment by kiriles90** [comment]
So, since the development branch has been updated to the latest NW.js version (https://github.com/popcorn-official/popcorn-desktop/commit/30ab2c5dd4177bd804a0a0ecab13e8bb77835763) but still lacks full proprietary audio codec support (the ffmpeg prebuild is better than nothing but is still lacking) I'm changing the title of this issue to reflect the new issue at hand and tagging this as a bug from feature that it was before the update.  

Also.. the old custom 0.44.5 NW.js version is now at [popcorn-official/popcorn-desktop/tree/development_old_nwjs](https://github.com/popcorn-official/popcorn-desktop/tree/development_old_nwjs) though this should be eventually removed once this issue here is resolved/closed. Last, Actions have been updated to reflect the change with now two workflows, one for new and one for old NW.js branches.

**Comment by Vhee22** [comment]
it seems the build artifacts are expired: https://github.com/popcorn-official/popcorn-desktop/actions/runs/5496485657
